src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -633,8 +633,20 @@ public bool IsFrozen
             }
             set
             {
-                isFrozenExplicitly = value;                                 
-                OnNodeModified();
+                isFrozenExplicitly = value;    
+                RaisePropertyChanged("IsFrozen");      
+                //If the node is Unfreezed then Mark all the downstream nodes as
+                // modified. This is essential recompiling the AST.
+                if (!value)
```

Comment:
If the freeze is false, then  set the downstream nodes to be modified. this helps when compiling the graph.